core/src/main/java/com/amplifyframework/auth/AuthException.java

```diff
@@ -191,4 +191,286 @@ public UnknownException(Throwable cause) {
+        private static final String RECOVERY_SUGGESTION =
+                "Please wait a while before re-attempting or increase the service limit.";
+
+        /**
+         * Default message/recovery suggestion with a cause.
+         * @param cause The original error.
+         */
+        public LimitExceededException(Throwable cause) {
+            super(MESSAGE, cause, RECOVERY_SUGGESTION);
+        }
+    }
+
+    /**
+     * Could not perform the action because password needs to be reset.
+     */
+    public static class PasswordResetRequiredException extends AuthException {
+        private static final long serialVersionUID = 1L;
+        private static final String MESSAGE = "Required to reset the password of the user.";
+        private static final String RECOVERY_SUGGESTION = "Reset the password of the user.";
+
+        /**
+         * Default message/recovery suggestion with a cause.
+         * @param cause The original error.
+         */
+        public PasswordResetRequiredException(Throwable cause) {
+            super(MESSAGE, cause, RECOVERY_SUGGESTION);
+        }
+    }
+
+    /**
+     * Could not perform the action because Amazon Cognito service cannot find the requested resource.
+     */
+    public static class ResourceNotFoundException extends AuthException {
+        private static final long serialVersionUID = 1L;
+        private static final String MESSAGE = "Could not find the requested online resource.";
+        private static final String RECOVERY_SUGGESTION = "Retry with exponential back-off";
+
+        /**
+         * Default message/recovery suggestion with a cause.
+         * @param cause The original error.
+         */
+        public ResourceNotFoundException(Throwable cause) {
+            super(MESSAGE, cause, RECOVERY_SUGGESTION);
+        }
+    }
+
+    /**
+     * Could not perform the action because user made too many failed attempts for a given action.
+     */
+    public static class FailedAttemptsLimitExceededException extends AuthException {
+        private static final long serialVersionUID = 1L;
+        private static final String MESSAGE = "User has made too many failed attempts for a given action.";
+        private static final String RECOVERY_SUGGESTION =
+                "Please check out the service configuration to see the condition of locking.";
+
+        /**
+         * Default message/recovery suggestion with a cause.
+         * @param cause The original error.
+         */
+        public FailedAttemptsLimitExceededException(Throwable cause) {
+            super(MESSAGE, cause, RECOVERY_SUGGESTION);
+        }
+    }
+
+    /**
+     * Could not perform the action because Amazon Cognito service encounters an invalid AWS Lambda response
+     * or encounters an unexpected exception with the AWS Lambda service.
+     */
+    public static class LambdaException extends AuthException {
```

Comment:
This type of exception is also implementation specific. (The Auth _category_ doesn't need to involved AWS Lambda, necessarily.) I'd recommend to remove this `LambdaException` entirely.